site/content/contribute/server/_index.md

```diff
@@ -20,10 +20,31 @@ The server consists of several different Go packages:
 
 * `api4` - Version 4 of the RESTful JSON Web Service
 * `app` - Logic layer for getting, modifying, and interacting with models
+* `services` - Set of packages providing functionality to the rest of the application, especially to the `app` package.
 * `cmd` - Command line interface
 * `einterfaces` - Interfaces for Enterprise Edition features
 * `jobs` - Job server and scheduling
 * `model` - Definitions and helper functions for data models
 * `store` - Storage layer for interacting with caches and databases
 * `utils` - Utility functions for various tasks
 * `web` - Serves static pages
+
+## New packages rules
+
+We are trying to follow certain rules about when is needed to create a new
+package or add functionality to the existing ones (Some of them can not fit
+with the current implementation because it is an ongoing effort).
+
+* Any integration with a third party tool should be implemented as an
+  independent service (Examples: Bleve, Cache, File backends).
+* Any well isolated chunk of code that has a very small public API and a lot of
+  code behind, should be separated into a service (Examples: Import/Export,
+  Tracing, Telemetry)
+* If you are adding logic that interact with differnt services, put the
+  information together, and give a result, probably you are adding logic to the
+  `app` package.
```

Comment:
I couldn't understand this sentence. By putting information together, do you mean gathering multiple service api calls into another struct/service like implementing a facade?

Reply:
I was thinking about things like, you get info from the store, then get a file from the files store, and provide that info to the end user. Or maybe use the email service to send a notification or the websockets to send some updates... The app layer is half bussiness logic and half glue code of the services (is the orchestrator of the data flowing in the app).

Reply:
I see, if the logic needs some `store` then we should supply it to the service with dependency injection and let the service handle the logic. Maybe we can emphasize this?

Reply:
I get your point, but I don't want to overcomplicate and over prescript things here. This are more guiding rules. I prefer to keep it simple, and try to be flexible in the application of the rules. For example, the services should receive the instances for the store injected, that is true, but that can happen from the App layer, or from the store layer, it depends on the service and how is it called.

Reply:
Sure, we can add these practices when we feel they are required. I was 0/5 on adding at the moment anyway. Thanks for the explanation.